Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.cpp

```diff
@@ -28,6 +28,8 @@
 
 namespace GradientSignal
 {
+    AZ_CLASS_ALLOCATOR_IMPL(EditorImageGradientComponentMode, AZ::SystemAllocator)
```

Comment:
Any reason that the combination of the  EditorImageGradientComponentMode.cpp/.h files are using the `AZ_CLASS_ALLOCATOR_DECL` and `AZ_CLASS_ALLOCATOR_IMPL` paradigm?

Reply:
Because its base class does:
https://github.com/o3de/o3de/blob/1c914f819c473ef4492360b02424e531727e846b/Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/EditorBaseComponentMode.h#L21-L26